connectors/utils.py

```diff
@@ -472,3 +472,19 @@ def get_pem_format(key, max_split=-1):
     key = " ".join(key.split("\n", max_split))
     key = " ".join(key.rsplit("\n", max_split))
     return key
+
+
+class PeriodicalChecker:
+    def __init__(self, interval):
+        self._interval = interval
+        self._start = None
+
+    def should_check(self):
+        if self._start is None:
+            self._start = time.time()
+
+        if time.time() - self._start < self._interval:
+            return False
+        else:
+            self._start = time.time()
+            return True
```

Comment:
Just looking at the implementation, seems like using `asyncio` Tasks can be a cleaner way to achieve this, for example like here https://stackoverflow.com/questions/37512182/how-can-i-periodically-execute-a-function-with-asyncio

What do you think?

Reply:
From the implementation, we have 2 intervals: 
1. JOB_REPORTING_INTERVAL - how often to report ingestion stats and send job heartbeat. This is hardcoded to 10 seconds in Ruby.
2. JOB_CHECK_INTERVAL - how often do we check whether the sync is done, it's supposed to run more frequently.
https://github.com/elastic/connectors-python/blob/f9fb26a08ac6f6d5df1e0318ee798b1c60f338d2/connectors/sync_job_runner.py#L118-L122

Not sure how the `asyncio.Task` solve this problem.

Reply:
What I mean is that it's possible to make an abstraction over `asyncio.Task` that repeats certain logic every X seconds. In the end current abstraction works and can be left as is (though `should_check` name is confusing too - should it be named differently, for example `period_elapsed` or `is_triggered`)

Reply:
I don't feel it's necessary to use `asyncio.Task`. In the [example](https://stackoverflow.com/questions/37512182/how-can-i-periodically-execute-a-function-with-asyncio) you posted, it was pushed to `create_task`, but it still needs to run `run_until_complete` to wait for it to complete. 

Same situation here that we need to wait for `self.elastic_server.done()` to proceed to `self._sync_done`. So I'd prefer to keep it in the foreground.

For `periodical_checker.should_check():`, I want to simulate [periodically](https://github.com/elastic/connectors-ruby/blob/21b8b6697007ae30f5d0555bd6271119e2155776/lib/core/sync_job_runner.rb#L244-L249) in ruby, but it seems there's no similar syntax of `yield`.

Regarding the name, the class is named `PeriodicalChecker`, which means it periodically (by configured interval) checks something, therefore the method name `should_check`. I think the method name should match the class name (`check` vs `checker`)

Reply:
I implemented it as a task: https://github.com/elastic/connectors-python/pull/515/commits/8f8854c9439b939342af7a9cb42984bc4a2ca2e1. But I haven't test it locally as I have issue starting the stack now.